### PR TITLE
openjdk8: fix build on macOS 10.7

### PR DESCRIPTION
**Pull request by catap**
#### Description

as nice side effect I've introduced support for xcode 14

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.7.5 11G63 x86_64
Xcode 4.6.3 4H1503

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
Just pushed a required fix for openjdk8-bootstrap; anyway I did it.

File: java/openjdk8/Portfile [hunk 1]
```diff
@@ -65,7 +79,20 @@ configure.args      --with-boot-jdk \
...
+                    --with-toolchain-type=clang
+} elseif { [string *gcc* ${configure.compiler}] } {
+    configure.args-append \
+                    --with-toolchain-type=gcc
+}
+
+build.pre_args-prepend \
+                    COMPILER_WARNINGS_FATAL=false
+                    # USE_PRECOMPILED_HEADER=0 \
+                    # QUIETLY= \
```
Comment (catap) [inline java/openjdk8/Portfile:93]: yay! I forgot this debug

File: java/openjdk8/Portfile [hunk 2]
```diff
@@ -29,11 +29,25 @@ checksums           rmd160  a5f2ee5ee3db4b42b5640c3364a026523a0af8c7 \
...
+                    0004-8030812-Change-the-solaris-DTrace-implementation-to-.patch \
+                    0005-os_perf_bsd-allow-to-compile-with-ZERO-CPU.patch \
+                    0006-Disable-C-11-warnings.patch \
+                    0007-Added-macosx-bin-zero-jvm.cfg.patch \
+                    0008-ExtendedOptionsImpl.c-define-TCP_KEEPINTVL-and-TCP_K.patch
+
+# NSFullSizeContentViewWindowMask is available since 10.10
+if {${os.platform} eq "darwin" && ${os.major} < 14} {
+    patchfiles-append \
+                    0009-Revert-8211301-macos-support-full-window-content-opt.patch
```
Comment (catap) [inline java/openjdk8/Portfile:44]: This patch can be removed by enforcing version of JDK before 292.

**Comment by catap** [comment]
For records. On 10.6 it fails as:
```
2022-10-01 01:35:43.344 java[12281:1d03] *** NSInvocation: warning: object 0x11478b468 of class 'ThreadUtilities' does not implement methodSignatureForSelector: -- trouble ahead
2022-10-01 01:35:43.345 java[12281:1d03] *** NSInvocation: warning: object 0x11478b468 of class 'ThreadUtilities' does not implement doesNotRecognizeSelector: -- abort
```
I have no idea how to hack bootstrap's java to make it works :(

**Comment by kakuhen** [comment]
This may not be relevant to building openjdk8 on Lion, but here is a GitHub repository that is able to produce a headless openjdk8u60 on my PowerMac G5. https://github.com/nilsvanvelzen/mac_ppc_openjdk8u60

**Comment by catap** [comment]
@kakuhen I haven't got access to PPC hardware to test it. Let me keep it as is.

The same technic can be applied to future ports, but before move on to the openjdk11 or PPC, I'd like to figure out how to bootstrap 10.6 and 10.5 intel.

**Comment by barracuda156** [comment]
@catap If possible, could you look into 10.6 and/or `ppc` case? Partial work was done here: https://github.com/macports/macports-ports/pull/14770
But it does not build still.
There is also this repo: https://github.com/nilsvanvelzen/mac_ppc_openjdk8u60

**Comment by catap** [comment]
@barracuda156 I may add it to todo list and try to look someday. But not today :)

**Comment by catap** [comment]
@mascguy any chance to review?

**Comment by mascguy** [comment]
> @mascguy any chance to review?

I reviewed a few days ago, but wasn't sure you were finished.  😉

**Review by mascguy — APPROVED** [review]

**Comment by barracuda156** [comment]
@mascguy @catap Portfile fails to get parsed:
```
Creating port index in /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports
Failed to parse file java/openjdk8/Portfile: unknown or ambiguous subcommand "*gcc*": must be bytelength, compare, equal, first, index, is, last, length, map, match, range, repeat, replace, reverse, tolower, totitle, toupper, trim, trimleft, trimright, wordend, or wordstart
```

**Comment by catap** [comment]
@barracuda156 may bad, I’ll fix it asap

**Pull request merged**